encoder.go

```diff
@@ -91,6 +92,22 @@ func avifEncoder(p1, p2 string, quality int) error {
 	if img.Metadata().Width > avifMax || img.Metadata().Height > avifMax {
 		return errors.New("WebP: image too large")
 	}
+
+	if config.EnableExtraParams {
+		imgHeightWidthRatio := float32(img.Metadata().Height) / float32(img.Metadata().Width)
+		if extraParams.Width > 0 && extraParams.Height > 0 {
+			err = img.Thumbnail(extraParams.Width, extraParams.Height, 0)
+			if err != nil {
+				return err
+			}
+		} else if extraParams.Width > 0 {
```

Comment:
what would happen if the user only provide with height? e.g. `1.jpg?height=1270`